symfony/plugins/orangehrmAttendancePlugin/lib/dao/AttendanceDao.php

```diff
@@ -470,50 +470,69 @@ public function searchAttendanceRecords($employeeIds = null, $employeementStatus
                 ->orderBy('a.punch_in_user_time DESC');
 
         if( $employeeIds != null){
-            
+
             if(is_array($employeeIds)){
                 $q->andWhereIn("e.emp_number", $employeeIds);
             } else {
                 $q->andWhere(" e.emp_number = ?", $employeeIds);
-            }            
+            }
         }
-        
-        if( $employeementStatus != null){           
+
+        if( $employeementStatus != null){
             $q->andWhere("e.emp_status = ?", $employeementStatus);
         } else {
             if($employeeIds <= 0){
                 $q->andWhere("(e.termination_id IS NULL)");
-            }            
+            }
         }
-        
+
         if( $subDivision > 0){
-            
+
             $companyService = new CompanyStructureService();
             $subDivisions = $companyService->getCompanyStructureDao()->getSubunitById($subDivision);
-           
+
             $subUnitIds = array($subDivision);
              if (!empty($subDivisions)) {
                 $descendents = $subDivisions->getNode()->getDescendants();
-               
-                foreach($descendents as $descendent) {                
+
+                foreach($descendents as $descendent) {
                     $subUnitIds[] = $descendent->id;
                 }
             }
-            
-            $q->andWhereIn("e.work_station", $subUnitIds);            
+
+            $q->andWhereIn("e.work_station", $subUnitIds);
         }
-        
-        if( $dateFrom != null){            
+
+        if( $dateFrom != null){
             $q->andWhere("a.punch_in_user_time >=?", $dateFrom);
         }
-        
+
         if( $dateTo != null){
             $q->andWhere("a.punch_out_user_time <=?", $dateTo);
         }
-      
+
         $result = $q->execute(array(), Doctrine::HYDRATE_SCALAR);
         return $result;
-      
+
+    }
+
+    public function getLastPunchRecordDetails($employeeId)
```

Comment:
This function return mix null and record and it always returns one record, better to rename as follows
```
public function getLatestPunchInRecord(int $employeeId)
```